### PR TITLE
[Ingest Manager] Make agent retry values for bootstraping configurable

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

This PR adds an option to configure values used in container command for GET/POST request to kibana. 

## Why is it important?

https://github.com/elastic/beats/issues/25063

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

cc @simitt

### CODE REVIEW

File: x-pack/elastic-agent/pkg/agent/cmd/container.go [hunk 1]
```diff
@@ -40,15 +41,19 @@ import (
 )
 
 const (
-	requestRetrySleep     = 1 * time.Second                  // sleep 1 sec between retries for HTTP requests
-	maxRequestRetries     = 30                               // maximum number of retries for HTTP requests
-	defaultStateDirectory = "/usr/share/elastic-agent/state" // directory that will hold the state data
+	requestRetrySleepEnv     = "KIBANA_REQUEST_RETRY_SLEEP"
+	maxRequestRetriesEnv     = "KIBANA_REQUEST_RETRY_COUNT"
+	defaultRequestRetrySleep = "1"                              // sleep 1 sec between retries for HTTP requests
```
Comment (ruflin) [inline x-pack/elastic-agent/pkg/agent/cmd/container.go:46]: Can we make this `1s` instead and have time parsing? This makes it possible to pass in any time values if needed and makes it more explicity.

File: x-pack/elastic-agent/pkg/agent/cmd/container.go [hunk 2]
```diff
@@ -70,6 +75,8 @@ The following actions are possible and grouped based on the actions.
   KIBANA_FLEET_USERNAME - kibana username to enable Fleet [$KIBANA_USERNAME]
   KIBANA_FLEET_PASSWORD - kibana password to enable Fleet [$KIBANA_PASSWORD]
   KIBANA_FLEET_CA - path to certificate authority to use with communicate with Kibana [$KIBANA_CA]
+  KIBANA_REQUEST_RETRY_SLEEP - specifies number of seconds of sleep taken when agent performs a request to kibana [default 1]
```
Comment (ruflin) [inline x-pack/elastic-agent/pkg/agent/cmd/container.go:78]: See above
```suggestion
  KIBANA_REQUEST_RETRY_SLEEP - specifies number of seconds of sleep taken when agent performs a request to kibana [default 1s]
```

**Review by ruflin — COMMENTED** [review]
One minor change request.

At one point we should discuss if these things should also be exposed through configs.

File: x-pack/elastic-agent/pkg/agent/cmd/container.go [hunk 3]
```diff
@@ -70,6 +75,8 @@ The following actions are possible and grouped based on the actions.
   KIBANA_FLEET_USERNAME - kibana username to enable Fleet [$KIBANA_USERNAME]
   KIBANA_FLEET_PASSWORD - kibana password to enable Fleet [$KIBANA_PASSWORD]
   KIBANA_FLEET_CA - path to certificate authority to use with communicate with Kibana [$KIBANA_CA]
+  KIBANA_REQUEST_RETRY_SLEEP - specifies sleep duration taken when agent performs a request to kibana [default 1s]
+  KIBANA_REQUEST_RETRY_COUNT - specifies number of retries agent performs when executing a request to kibana [default 30]
```
Comment (simitt) [inline x-pack/elastic-agent/pkg/agent/cmd/container.go:79]: Can you please also add support for this to the `kibanaConfig` https://github.com/elastic/beats/pull/25163/files#diff-1f8e1e2079524931fa8f3cf121667a25baa2b998557136500d5e7596b46e9187L781?

**Review by simitt — APPROVED** [review]
Havent't tested yet, but LGTM

**Comment by urso** [comment]
In general LGTM.

Should we consider exponential backoff (see: [libbeat/common/backoff](https://github.com/elastic/beats/tree/master/libbeat/common/backoff))?

**Comment by simitt** [comment]
> Should we consider exponential backoff (see: libbeat/common/backoff)?

Can this be done in a follow up PR?

**Comment by michalpristas** [comment]
i think blake did not go with exponential because we're just waiting for kibana to be online. we want result as soon as possible and we're not spamming kibana with request as it does not perform any work in this case.

**Pull request merged**